keccak256/src/gates/state_conversion.rs

```diff
@@ -0,0 +1,186 @@
+use halo2::{
+    circuit::Layouter,
+    plonk::{Advice, Column, ConstraintSystem, Error},
+};
+
+use crate::gates::base_conversion::BaseConversionConfig;
+use crate::gates::tables::BaseInfo;
+use pairing::arithmetic::FieldExt;
+use std::convert::TryInto;
+
+#[derive(Debug, Clone)]
+pub struct StateBaseConversion<F> {
+    bi: BaseInfo<F>,
+    bccs: [BaseConversionConfig<F>; 25],
+    state: [Column<Advice>; 25],
+}
```

Comment:
The base conversion as mentioned in https://github.com/appliedzkp/zkevm-circuits/issues/211 can be used **conditionally** inside of Mixing. 

Therefore, I suggest a similar configuration approach as the one used in `IotaB9` in #144 .

The gate has to be:
- Conditionally enabled due to a flag.
- Be able to be called inside and outside of Mixing (outside the flag is useless).

Reply:
@CPerezz I added flags on StateBaseConversion and BaseConversionConfig. The flag has these properties:

1. flag is binary
2. parent flag can enable or disable the children's polys.

- The flag is a column, all cells are copied from the parent flag.
- The parent flag is assumed to be constrainted to be 0 or 1. (full fill property 1)
- All polys multiply the flag.  (full fill property 2)

Reply:
Makes complete sense! I like it!!!